### ISSUE TITLE
Generate Room Config und Multi Map

### ISSUE THREAD

**Issue by littleyoda** [humbertogontijo/homeassistant-roborock#197]
Hi

Generate Room Config unfortunately doesn't work quite right for me.
Only 4 rooms are created, whereas there are 5 rooms in the app.

- I checked the predefined_selections. There are only 4 entries
- You can also see the problem on the map

App:
![grafik](https://user-images.githubusercontent.com/6692041/222890460-b9fa7147-743e-469e-8839-11171da786ee.png)

Home-Assistant:
![grafik](https://user-images.githubusercontent.com/6692041/222890485-1d0e90fd-42fe-45fe-b1c7-af1c11853494.png)

**Comment by PiotrMachowski** [comment]
> Wrong repository. Look over here [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#518](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/issues/518)

I don't think it is related to my card. It looks like there *might* be 4 rooms instead of 5, or maybe just 2 of them use the same color (but in this case they still should be correctly distinguished when generating rooms in card)

**Comment by littleyoda** [comment]
My understanding is that [homeassistant-roborock](https://github.com/humbertogontijo/homeassistant-roborock) provides the data and [lovelace-xiaomi-vacuum-map-card/](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/issues/518) virtualises the information.

Since only 4 rooms appear in the "predefined_selections", I assumed that there was an error in this integration.
It should be noted that it works for one map. On the second map, the number of rooms is not correct.

I don't know if it matters. The IDs of the rooms are not unique. Some rooms on the first and second floor have the same IDs.

**Comment by littleyoda** [comment]
How can I help to solve this problem?

Some additional screenshots:
Groundfloor
![image](https://user-images.githubusercontent.com/6692041/229341124-aa51c0dd-a9e6-472d-8848-d228849c3036.png)

First Floor
![image](https://user-images.githubusercontent.com/6692041/229341134-cbf1b15f-c1a3-47b1-ba3a-8254c018d89e.png)

**Comment by silelmot** [comment]
I have the same problem. Rooms are only generated for the active room when configuring the lovelance-template.
Till this is resolved in the code, is there a way to get around this? Exporting the map data and import based on loaded map?

**Comment by littleyoda** [comment]
Any updates on this bug?
Or hints how to fix it?

**Comment by mebi2024** [comment]
I had the same problem. No rooms are displayed due to the point "reading the roomcards".
Maybe this is a workaround:
My menus are in german so try "Entwicklerwerkzeuge" => "Entitäten"
in my case I search directly in the table header for "camera.roborock_s7_map".
On the right there are attributes and if everything is displayed in the cell "Attribute" I found the code snipped below, which gives me at least the room numbers. You have to scroll down - really down! - nearly up to the end in my case. However, I have to guess the names of the rooms, but I have now the used room numbers!

room_numbers: 
'16': null
'17': null
'18': null
'19': null
'20': null
'21': null
'22': null
'23': null
'24': null
'25': null

rooms: 
'16':
  x0: 16200
  y0: 22800
  x1: 19250
  y1: 26350
'17':
  x0: 23750
  y0: 26450
  x1: 30150
  y1: 32800
'18':
etc.
